manifests/key_pair.pp

```diff
@@ -0,0 +1,31 @@
+define certs::key_pair (
+  $source_dir,
+  $key_file,
+  $cert_file,
+  $key_ensure  = 'present',
+  $key_owner   = undef,
+  $key_group   = undef,
+  $key_mode    = undef,
+  $cert_ensure = 'present',
+  $cert_owner  = undef,
+  $cert_group  = undef,
+  $cert_mode   = undef,
```

Comment:
Can we get into the habit of always adding data types to new parameters?

I also think we should set a mode by default for both types. This reduces the risk of accidentally exposing private keys.